### DISCUSSION TITLE
Verify JWT token outside of NextJS

### DISCUSSION THREAD

**Discussion by ethanplee14** [nextauthjs/next-auth#2986] [Help]
I have a NextJS application which connects the front end to a separate stand alone socketio server so I can avoid creating a custom NextJS server. I am having trouble verifying the JWT token outside of NextJS/NextAuth. I configured nextauth with
```
    secret: process.env.JWT_SECRET,
    jwt: {
        secret: process.env.JWT_SECRET
    }
```
and then I try and use the jsonwebtoken module to verify the token with my JWT_SECRET but verification fails. Is there any way to do this?

**Comment by balazsorban44** [comment]
I think so, yes, check the settings here:
 https://github.com/nextauthjs/next-auth/blob/main/src/lib/jwt.js

you might be better off if on login, your identity provider sends you the access token and you use that, instead of trying to use the one next-auth creates.

**Reply by ethanplee14** [reply]
Meaning that once the identity provider sends me the access token, i can parse the information from it and create a new token to send and verify on my stand alone server?
